File: Class_06_Frontend_Interviews_And_Merge_Sort/practice.py
#[3,2,3] -> 3
#[2,2,1,1,1,2,2] -> 2
input = [3,2,3]
def majorityElement(arr):
    returnDict = {}
    for element in arr:
        if element in returnDict:
            returnDict[element] += 1
        else:
            returnDict[element] = 1
    for key in returnDict:
        if returnDict[key] > (len(arr)//2):
            return key

print(majorityElement(input))

# def majorityElementMergeSort(arr):
#     merge2(arr, 0, len(arr) -1)
#     return arr

# def merge2(arr, first, last):

File: Class_08_Dictionary_Problems_Medium/practice.js
// given an array of size n, return the majority element

input = [3,2,3] // returns 3
input2 = [2,2,2,1,1,2] // returns 2

function majorityElement(arr) {
    output = {}
    for (num of arr) {
        if (num in output) {
            output[num] ++
        } else {
            output[num] = 1
        }
    }
    for (num of arr) {
        if ((Math.floor(output[num])) >= (Math.floor(arr.length / 2))) {
            return num
        }
    }
}

console.log(majorityElement(input))
console.log(majorityElement(input2))